src/plugin/class-storage.php

```diff
@@ -3,44 +3,88 @@
+	private string $crawler_data_post_type_name_plural = 'Crawler URLs';
 
 	private array $custom_post_types_supports = array( 'title', 'editor', 'custom-fields' );
 
-	public function __construct( string $post_type ) {
-		$this->post_type      = $post_type;
-		$this->post_type_name = ucwords( str_replace( '_', ' ', $post_type ) );
+	public function __construct( string $liberated_data_post_type, string $crawler_data_post_type ) {
+		$this->liberated_data_post_type      = $liberated_data_post_type;
+		$this->liberated_data_post_type_name = ucwords( str_replace( '_', ' ', $liberated_data_post_type ) );
+		$this->crawler_data_post_type        = $crawler_data_post_type;
 
 		add_action( 'init', array( $this, 'register_post_types' ) );
 	}
 
-	private function get_singular_name(): string {
-		return $this->post_type_name;
-	}
+	public function register_post_types(): void {
+		register_post_type(
+			$this->liberated_data_post_type,
+			array(
+				'public'              => false,
+				'exclude_from_search' => true,
+				'publicly_queryable'  => false,
+				'show_in_rest'        => true,
+				'show_ui'             => true,
+				'show_in_menu'        => WP_DEBUG,
+				'menu_icon'           => 'dashicons-database',
+				'supports'            => $this->custom_post_types_supports,
+				'labels'              => $this->get_post_type_registration_labels(
+					$this->liberated_data_post_type_name,
+					$this->liberated_data_post_type_name
+				),
+				'rest_base'           => $this->liberated_data_post_type,
+			)
+		);
 
-	private function get_plural_name(): string {
-		return $this->post_type_name;
-	}
+		register_post_type(
+			$this->crawler_data_post_type,
+			array(
+				'public'              => false,
+				'exclude_from_search' => true,
+				'publicly_queryable'  => false,
+				'show_in_rest'        => false,
+				'show_ui'             => true,
+				'show_in_menu'        => WP_DEBUG,
+				'menu_icon'           => 'dashicons-editor-ul',
+				'supports'            => array( '' ), // has to be empty string array, otherwise title and content support comes in by default
+				'labels'              => $this->get_post_type_registration_labels(
+					$this->crawler_data_post_type_name,
+					$this->crawler_data_post_type_name_plural
+				),
+				'rest_base'           => $this->liberated_data_post_type,
+			)
+		);
 
-	public function register_post_types(): void {
-		$name        = $this->get_singular_name();
-		$name_plural = $this->get_plural_name();
-
-		$args = array(
-			'public'              => false,
-			'exclude_from_search' => true,
-			'publicly_queryable'  => false,
-			'show_in_rest'        => true,
-			'show_ui'             => true,
-			'show_in_menu'        => WP_DEBUG,
-			'menu_icon'           => 'dashicons-database',
-			'supports'            => $this->custom_post_types_supports,
-			'labels'              => $this->get_post_type_registration_labels( $name, $name_plural ),
-			'rest_base'           => $this->post_type,
+		register_post_status(
```

Comment:
It feels similar to the [WP_Stream_Importer](https://github.com/WordPress/wordpress-playground/blob/571128bde1c2caf586d0462904952434ddaad30a/packages/playground/data-liberation/src/import/WP_Stream_Importer.php#L31) that fetches the assets and incrementally inserts WordPress entities such as posts, pages, tags etc. It seems like we're duplicating efforts – let's find a way to converge and build a more general solution that would serve this extension's use case.